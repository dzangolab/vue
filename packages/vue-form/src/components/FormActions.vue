<template>
  <div
    :class="`align-${alignment} direction-${flowDirection}`"
    class="form-actions"
  >
    <slot>
      <slot name="cancelButton">
        <button
          :disabled="loading"
          class="cancel-button"
          type="button"
          @click="onCancel"
        >
          {{ cancelLabel }}
        </button>
      </slot>
      <slot name="submitButton">
        <button
          :disabled="loading"
          class="submit-button"
          type="submit"
          @click="onSubmit"
        >
          {{ submitLabel }}
        </button>
      </slot>
    </slot>
  </div>
</template>

<script lang="ts">
export default {
  name: "FormActions",
};
</script>

<script setup lang="ts">
defineProps({
  alignment: {
    default: "right",
    type: String,
  },
  cancelLabel: {
    default: "Cancel",
    type: String,
  },
  flowDirection: {
    default: "horizontal",
    type: String,
  },
  loading: {
    default: false,
    type: Boolean,
  },
  submitLabel: {
    default: "Submit",
    type: String,
  },
});

const emit = defineEmits(["cancel", "submit"]);

const onCancel = () => emit("cancel");

const onSubmit = () => emit("submit");
</script>

<style lang="css" scoped>
.form-actions {
  display: flex;
  gap: var(--form-field-gap);
  margin-bottom: 2rem;
  width: 100%;
}

.form-actions.align-center {
  justify-content: center;
}

.form-actions.align-left {
  justify-content: left;
}

.form-actions.align-right {
  justify-content: right;
}

.form-actions.align-filled {
  justify-content: space-between;
}

.form-actions.direction-horizontal {
  flex-direction: row;
}

.form-actions.direction-vertical {
  flex-direction: column;
}

button {
  border-radius: 0.5rem;
  cursor: pointer;
  float: right;
  font-weight: 700;
  padding: 0.75rem 1.25rem;
  width: max-content;
}

.form-actions .submit-button {
  background: var(--dz-primary-color, #007aff);
  border: none;
  color: #fff;
}

.form-actions .cancel-button {
  border: 1px solid var(--button-border-color, #000);
}
</style>
