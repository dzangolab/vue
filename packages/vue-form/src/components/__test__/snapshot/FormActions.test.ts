import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import FormActions from "../../FormActions.vue";

describe("FormActions", () => {
  it("matches snapshot", () => {
    const wrapper = mount(FormActions, {
      global: {
        stubs: {
          ButtonElement: true,
        },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
