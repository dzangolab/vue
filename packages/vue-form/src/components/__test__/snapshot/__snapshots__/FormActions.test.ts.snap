// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormActions > matches snapshot 1`] = `
<div
  class="form-actions"
  data-alignment="right"
  data-direction="horizontal"
  data-reverse="false"
>
  
  
  
  <button-element-stub
    aria-label="button"
    disabled="false"
    id="submit"
    label="Submit"
    loading="false"
    rounded="false"
    severity="primary"
    size="medium"
    type="submit"
    variant="filled"
  />
  
  
  <button-element-stub
    aria-label="button"
    disabled="false"
    id="cancel"
    label="Cancel"
    loading="false"
    rounded="false"
    severity="secondary"
    size="medium"
    type="button"
    variant="outlined"
  />
  
  
  
</div>
`;
